.github/workflows/build.yml

```diff
@@ -0,0 +1,116 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+      - gh-actions # TODO: remove this.
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron: '0 4 * * *' # Runs every day at 4am: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule
+
+jobs:
+
+  build:
+    name: ${{ matrix.os }}, Node.js v${{ matrix.node }}, Python ${{ matrix.python }}
+
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        node: ['12.x']
+        python: ['2.x']
```

Comment:
Based on this https://github.com/eclipse-theia/theia/pull/8780/files#r530476212, we can also pin the Python and Node.js version. I do not have a strong opinion on this. For the record: the Travis-based build uses a floating version of Node.js.

Reply:
yarn is also automatically provided by the runners so it may have different behaviour over time.
It depends where you want to put the cursor of 'validate automatically against latest' and 'use pre-defined versions'
maybe some jobs could check against latest of everything and the one that publishes should use pre-defined versions